### ISSUE TITLE
ScatterGL attributes.js is missing 'spline' shape

### ISSUE THREAD

**Issue by brian428** [plotly/plotly.js#3229]
It looks like "spline" is missing from the ScatterGL attributes file for line shape options: https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/attributes.js#L51

**Comment by etpinard** [comment]
Part of https://github.com/plotly/plotly.js/issues/130, but yeah this now deserves an issue of its own.

**Comment by brian428** [comment]
Oh, was "spline" never a valid option here?

**Comment by etpinard** [comment]
No. `1.42.0` added `line.shape` to `scattergl`, but w/o the `'spline'` value.

**Comment by Akhliskun** [comment]
I was reading #130 but I wonder if there is anything more specific to watch about the progress on "spline".
(in the meantime I'm subscribed to this for updated)

**Comment by Sagaulf** [comment]
Any news on this? 
Using plotly express line chart with dash. The graph is initially loaded with the `'spline'` format but on a callback it fails due to `scattergl.line` not having `'spline'` as a shape property

**Comment by ZeroCool2u** [comment]
Same issue here as @Sagaulf This seems like a bug, because the Plotly Express docs don't mention this constraint at all.

**Comment by skoeb** [comment]
Also wondering if there is a fix for this... it seems like in plotly express 'spline' works with `svg` as the `render_mode` but not `auto` in some circumstances.

**Comment by patryk-oleniuk** [comment]
I have the same problem. 
Furthermore, depending on the amount of data (if `render_mode=auto`) the render mode is switched from `svg` to `webgl`. So I currently have the situation when the `spline` sometimes works, sometimes does not...

**Comment by valankar** [comment]
> Furthermore, depending on the amount of data (if `render_mode=auto`) the render mode is switched from `svg` to `webgl`. So I currently have the situation when the `spline` sometimes works, sometimes does not...

I just spent quite awhile trying to figure out why I was getting an exception depending on how much data used, and it turned out to be this. Please fix this.

**Comment by pedramardakani** [comment]
Not a huge pain yet. But seems like this feature could use some attention if not too much effort.

**Comment by genseirin** [comment]
I would be interested if this is on the roadmap for a new version.

**Comment by robertclaus** [comment]
Thank you for bringing this up again. We likely won't get to this issue in the near future since we are working on a number of other issues, but would be very open to a community PR to resolve it.
